packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -291,25 +288,45 @@ export class ChainService implements ChainServiceInterface {
     subscriptions.map(s => s.unsubscribe());
   }
 
+  private async getInitialHoldings(contract: Contract, channelId: string): Promise<DepositedEvent> {
+    const holding = BN.from(await contract.holdings(channelId));
+
+    return {
+      type: Deposited,
+      channelId,
+      assetHolderAddress: makeAddress(contract.address),
+      amount: BN.from(holding),
+    };
+  }
+
+  private async waitForConfirmations(event: Event | undefined): Promise<void> {
+    if (event) {
+      await (await event.getTransaction()).wait(blockConfirmations + 1);
```

Comment:
Can you explain this logic in a comment to highlight what's going on?

It seems like the core change in this PR, and while it is possible to infer what's going on, it might not be obvious. 

Something like
```suggestion
      // `tx.wait(n)` resolves after n blocks are mined that include the given transaction `tx`
      // See https://docs.ethers.io/v5/api/providers/types/#providers-TransactionResponse
      await (await event.getTransaction()).wait(blockConfirmations + 1);
```